.github/workflows/build.yml

```diff
@@ -19,17 +19,21 @@ env:
 
 jobs:
   get-go-version:
+    # Inspired by envconsul -- https://github.com/hashicorp/envconsul/blob/bcb270fdc53e1273b3010d51c02fcf2e67d830d0/.github/workflows/build.yml#L18
     name: "Determine Go toolchain version"
     runs-on: ubuntu-latest
     outputs:
       go-version: ${{ steps.get-go-version.outputs.go-version }}
     steps:
       - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # 5.1.0
```

Comment:
The first attempt to remove `.go-version` was https://github.com/hashicorp/terraform-provider-cloudinit/pull/282/files

I have more confidence in this one.